src/Raven.Server/Documents/Indexes/Static/StaticMapIndex.cs

```diff
@@ -31,34 +33,15 @@ private static dynamic ToDynamic(object value)
             return expando as ExpandoObject;
         }
 
-        public override IEnumerable<Document> EnumerateMap(IEnumerable<Document> documents, string collection, TransactionOperationContext indexContext)
+        public override IEnumerable<object> EnumerateMap(IEnumerable<Document> documents, string collection, TransactionOperationContext indexContext)
         {
             var funcs = _compiled.Maps[collection];
 
             if (funcs.Length == 1)
             {
-                Document current = null;
-                foreach (var doc in funcs[0](documents.Select(x =>
+                foreach (var doc in funcs[0](documents.Select(x => new DynamicDocumentObject(x)))) // TODO [ppekrol] can we reuse DynamicDocumentObject
```

Comment:
You can reuse this, yes.
And it is better to avoid the `.Select()` here.
Create a single `DynamciDocumentObject`, which will point to a blittable doc.
And on each loop iteration, just return it after setting the new blittable doc.
For fun, we can also dispose the previous one immediately, instead of waiting for the end of the batch.

Also, why is there special behavior for multiple functions? We can just iterate them all in one shot.